### PR TITLE
[KS] ProgressMeter Bugfix transparancy and responsibility

### PR DESCRIPTION
**Pull request by qdittrich**
This bugfix for the ProgressMeter KitchenSink component is step one of the bugfix- / feature-bundle, mentioned in the ticket below.
This bugfix has only effect on the "mini" type of the component. The  wording "mini" does not longer refer to the size, but is meant as an shortcut for "minimalistic" as told in the modified description.

https://mantis.ilias.de/view.php?id=23654

### CODE REVIEW

**Comment by Amstutz** [comment]
@qdittrich thx for this fix. However, one test is failing, please fix:
image

**Comment by qdittrich** [comment]
I'm currently not able to properly fix the PHPUnit test, because since any update some time ago, I cannot get the PHPUnit tests running on my machine anymore. Even reinstalling PHPUnit and composer does not work.

I only changed one file (I added some spaces at a string) and now Travis throws lots of errors on other files, I never touched. The last time there was an similiar behavior about that "Dropzone" components; do you remember @Amstutz ?
However, all "ProgressMeter" tests pass. The last failure was because of missing spaces inside to parts of the test string. I added these spaces and now this test seems to pass.

File: src/UI/templates/default/Chart/chart.less [hunk 1]
```diff
@@ -524,4 +466,55 @@
...
+  &.il-chart-progressmeter-mini {
+    &:before, &:after {
+      content: none;
+    }
+  }
+
+  .il-chart-progressmeter-mini-container {
+    position: relative;
+    min-width: 20px;
+    max-width: 250px;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less]: Where do those values come from? How are they calculated/determined?

File: src/UI/templates/default/Chart/chart.less [hunk 2]
```diff
@@ -524,4 +466,55 @@
...
+    width: 100%;
+    height: auto;
+
+    .il-chart-progressmeter-viewbox {
+      display: block;
+      max-width: 80%;
+      min-width: 20px;
+      max-height: 250px;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less]: Same values as above, introduce variables.

File: src/UI/templates/default/Chart/chart.less [hunk 3]
```diff
@@ -524,4 +466,55 @@
...
+
+      .il-chart-progressmeter-circle-bg {
+        stroke: @il-chart-progressmeter-background;
+        fill: none;
+        stroke-width: 5;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less]: Definitely introduce a variable for this one (e.g. il-chart-progressmeter-mini-stroke-width)

File: src/UI/templates/default/Chart/chart.less [hunk 4]
```diff
@@ -524,4 +466,55 @@
...
+        stroke-linecap: round;
+      }
+      .il-chart-progressmeter-circle {
+        fill: none;
+        stroke-width: 5;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less]: Again the variable

File: src/UI/templates/default/Chart/chart.less [hunk 5]
```diff
@@ -524,4 +466,55 @@
...
+      .il-chart-progressmeter-circle.no-success {
+        stroke: @il-chart-progressmeter-bar-no-success;
+      }
+      .il-chart-progressmeter-circle.success {
+        stroke: @il-chart-progressmeter-bar-success;
+      }
+      .il-chart-progressmeter-needle-mini {
+        stroke: @il-chart-progressmeter-mini-marker-color;
+        fill: none;
+        stroke-width: 4px;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less]: Make this dependend from the above variable or maybe event introduce a new one.

**Comment by Amstutz** [comment]
Hi @qdittrich Thx for fixing the failing tests. The reason travis was failing in other tests was kind of "bad luck" from your side. You committed the fix of your tests while other tests where broken due to work of another dev (see build history in travis). This tests have been fixed by the according dev. in the meanwhile. All I needed to do was to hit rerun for your PR in travis (I am not sure if you can do that):
image

and all the lights for your test did turn green. Since this is a regular bugfix, there would be no issue with merging this now. However there are some todos. Please fix them before we can proceed:

- [ ] You changed the description of your component. This is good, thx. However, such changes (especially fundemantal ones like yours) need to be approved by the JF). Take this out for the moment and open a new PR with just the change of the description so we can merge the technical fix know and discuss your proposal of the semantic change later in the JF with a new PR (propably together with the other changes you are intending to perform for the other progress meter.
- [ ] Currenlty the new mini version does not look right in the examples (tested on safari and chrome). In my case they look like this:
image

- [ ] Some minor points with the less (see my comments).

[Edit:] I just found the reason, the less was not recompiled. So the second point is easily fixed by just recompiling less to include your changes in the css and commit.

**Review by Amstutz — CHANGES_REQUESTED** [review]

**Comment by qdittrich** [comment]
First of all: Thank you for your comments and your help! 👍 

I will revert the component description and set it in the "feature" PR again, as you mentioned above. I did not know that changes in this place need a JF decision.

About the styling: I will introduce some new variables.
For explanation about the sizes:
The min-width and max-width / max-height are set because it seemed to look a bit distorted if it gets smaller then 20px or bigger then 250px. Also, if it gets smaller then 20px, you can not clearly see your progress anymore because it is to small (at least for my eyes).

**Comment by qdittrich** [comment]
Hi @Amstutz 
I updated the less files, introduced some variables and recompiled the delos less. Also I reverted the component description and merged the trunk, so the PHPUnit tests should pass now.

**Comment by Amstutz** [comment]
Hi @qdittrich almost ready to merge ;-). There is a new issue, double check the CSS file, I believe you committed a lot more changes than should originate from you less. This might be due to your configuration of your lessc compiler. Nervertheless, you should only commit css changes that do origin from you own less.

**Comment by Amstutz** [comment]
Thx a lot for this fix!

**Pull request merged**